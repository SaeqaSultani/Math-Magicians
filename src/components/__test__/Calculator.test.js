import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../calculator';

it('Check for any changes to the component', () => {
  const tree = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should return 8  for (2+2) * 2', () => {
  render(
    <Calculator />,
  );
  const display = screen.getByTestId('display');
  const two = screen.getByText('2');
  const plus = screen.getByTestId('+');
  const multiply = screen.getByText('x');
  const equal = screen.getByText('=');
  fireEvent.click(two);
  fireEvent.click(plus);
  fireEvent.click(two);
  fireEvent.click(multiply);
  fireEvent.click(two);
  fireEvent.click(equal);
  expect(display).toHaveTextContent('8');
});
