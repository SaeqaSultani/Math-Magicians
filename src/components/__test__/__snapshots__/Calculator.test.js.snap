// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for any changes to the component 1`] = `
<div
  className="div-container"
>
  <p
    className="text-p"
  >
    Lets do some math!
  </p>
  <div
    className="calculat"
  >
    <div
      className="input"
      data-testid="display"
    />
    <div
      className="parent"
    >
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        %
      </button>
      <button
        className="button-color"
        data-testid="+"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        9
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="submit"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        6
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        3
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
      <button
        className="button-zero"
        onClick={[Function]}
        type="submit"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        .
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
